File: FTTradeDemo/FTTradeDemo3/FTTradeDemo3/Card.m
//
//  Card.m
//  FTTradeDemo3
//
//  Created by admin on 2022/4/8.
//

#import "Card.h"

@implementation Card
- (instancetype)initWithCardTurnoffContentUrl:(NSString *)cardTurnoffContentUrl CardTitle:(NSString *)cardTitle CardBottomTitle:(NSString *)cardBottomTitle {
    self = [super init];
    
    if(self) {
        self.cardTurnoffContentUrl = cardTurnoffContentUrl;
        self.cardTitle = cardTitle;
        self.cardContentUrl = @"card_before.png";
        self.cardBottomTitle = cardBottomTitle;
        self.isSelected = NO;
    }
    return self;
}
- (void)turnOffCard {
    if(self.isSelected) {
        self.isSelected = !self.isSelected;
        self.cardContentUrl = @"card_before.png";
    } else {
        self.isSelected = !self.isSelected;
        self.cardContentUrl = self.cardTurnoffContentUrl;
    }
}

@end

File: FTTradeDemo/FTTradeSwiftDemo4/FTTradeSwiftDemo4/Card.swift
//
//  Card.swift
//  FTTradeSwiftDemo4
//
//  Created by admin on 2022/4/18.
//

import Foundation

class Card {
    
    var cardTitle : String
    var cardContentUrl : String
    var cardTurnOffContentUrl : String
    var isSelected : Bool
    
    init(cardTurnOffContentUrl: String, cardTitle: String ) {
        self.cardTurnOffContentUrl = cardTurnOffContentUrl
        self.cardTitle = cardTitle
        self.cardContentUrl = "card_before.png"
        self.isSelected = false
    }
    
    func turnOffCard() {
        if(self.isSelected) {
            self.isSelected = !self.isSelected
            self.cardContentUrl = "card_before.png"
        } else {
            self.isSelected = !self.isSelected
            self.cardContentUrl = self.cardTurnOffContentUrl
        }
    }
}
